lib/semian/redis.rb

```diff
@@ -16,11 +16,17 @@ def initialize(semian_identifier, *args)
   ResourceBusyError = Class.new(SemianError)
   CircuitOpenError = Class.new(SemianError)
 
-  # This memoized alias is necessary because during a `pipelined` block
-  # the client is replaced by an instance of `Redis::Pipeline` and there is
-  # no way to access the original client.
-  def semian_resource
-    @semian_resource ||= @client.semian_resource
+  attr_reader :semian_resource
+
+  alias_method :_original_initialize, :initialize
+
+  def initialize(*args, &block)
+    _original_initialize(*args, &block)
```

Comment:
I sticked with the file stile. I'm not 100% sure why we didn't use prepend in semian. IIRC it's because prepeding a method defined in C and calling super used to make MRI segfault.

I'd be happy to convert semian to prepend, but not in this PR though.